### ISSUE TITLE
False positive no-member with wtforms BaseForm.populate_obj

### ISSUE THREAD

**Issue by bodik** [PyCQA/pylint#4643]
Hello,

during package bump in my project (sner4) I've came across perhaps false positive `no-member` related to wtforms package for function `BaseForm.populate_obj()` (https://github.com/wtforms/wtforms/blob/master/src/wtforms/form.py#L73)

According to minimal test case the issue is present since pylint 2.5.0, with last good version pylint 2.4.4

Any analysis or advice would be appreciated.

### Steps to reproduce
```py
"""
pylint vs wtforms 'populate_obj' member (no-member) minimal test example

python3 -m venv venv
venv/bin/pip install flask_wtf


venv/bin/pip install 'pylint==2.4.4'
venv/bin/pylint test.py
>> OK


venv/bin/pip install 'pylint==2.9.1'
venv/bin/pylint test.py
>> ERROR test.py:35:4: E1101: Instance of 'MyForm' has no 'populate_obj' member (no-member)
"""

import inspect

from flask import Flask
from flask_wtf import FlaskForm
from wtforms import IntegerField


class MyForm(FlaskForm):
    """example form"""
    avalue = IntegerField('avalue')


class MyDto:
    """example dto"""
    def __init__(self):
        self.avalue = 0


def main():
    """main"""

    app = Flask(__name__)
    app.config['SECRET_KEY'] = 'akey'

    with app.test_request_context():
        mydto = MyDto()
        myform = MyForm()
        myform.populate_obj(mydto)


    print(inspect.getmro(MyForm))
    print(MyForm.populate_obj)
    print(callable(MyForm.populate_obj))


if __name__ == '__main__':
    main()
```

### Current behavior

```
ERROR test.py:35:4: E1101: Instance of 'MyForm' has no 'populate_obj' member (no-member)
```

### Expected behavior

no `no-member` error

### pylint --version output

```
pylint 2.9.1
astroid 2.6.2
Python 3.7.3 (default, Jan 22 2021, 20:04:44) 
[GCC 8.3.0]
```

Additional dependencies:
```
Flask-WTF==0.15.1
WTForms==2.3.3
```

**Comment by cdce8p** [comment]
This might be related to #4278

**Comment by timmartin** [comment]
I had a quick look at this, and it looks to me like it's related to the way `Form` is implemented:

    class Form(with_metaclass(FormMeta, BaseForm)):
        ...

where `with_metaclass` is a Python 2 compatibility feature. The problem seems to go away in wtforms 3.0, which apparently drops Python 2 compatibility.

I see that Astroid already has brain code for `six.with_metaclass`, and I think wtforms is essentially reimplementing a limited version of what `six.with_metaclass` does. I guess we can copy this existing logic.

**Comment by bodik** [comment]
wtforms 3.0.0 released on 2021-11-07 does not produce E1101 with latest pylint. thanks to all for analysis and help.
